Algorithm.CSharp/IndicatorSelectorsWorkWithDifferentOptions.cs

```diff
@@ -0,0 +1,166 @@
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// This regression algorithm tests that we can use selectors in the indicators
+    /// that need quote data
+    /// </summary>
+    public class IndicatorSelectorsWorkWithDifferentOptions: QCAlgorithm, IRegressionAlgorithmDefinition
+    {
+        private List<Indicator> _equityIndicators;
+        private Indicator _optionIndicator;
+        private Symbol _equity;
+        private Symbol _option;
+        private bool _quoteBarsFound;
+        private bool _tradeBarsFound;
+
+        public override void Initialize()
+        {
+            SetStartDate(2013, 06, 07);
+            SetEndDate(2013, 11, 08);
+
+            _equity = AddEquity("SPY", Resolution.Minute).Symbol;
+            _option = AddOption("NWSA", Resolution.Minute).Symbol;
+            _option = QuantConnect.Symbol.CreateOption("NWSA", Market.USA, OptionStyle.American, OptionRight.Put, 33, new DateTime(2013, 07, 20));
+            AddOptionContract(_option, Resolution.Minute);
+
+            _equityIndicators = new List<Indicator>()
+            {
+                Identity(_equity, Resolution.Minute, Field.BidClose, "Bid.Close."),
+                Identity(_equity, Resolution.Minute, Field.BidOpen, "Bid.Open."),
+                Identity(_equity, Resolution.Minute, Field.BidLow, "Bid.Low."),
+                Identity(_equity, Resolution.Minute, Field.BidHigh, "Bid.High."),
+                Identity(_equity, Resolution.Minute, Field.AskClose, "Ask.Close."),
+                Identity(_equity, Resolution.Minute, Field.AskOpen, "Ask.Open."),
+                Identity(_equity, Resolution.Minute, Field.AskLow, "Ask.Low."),
+                Identity(_equity, Resolution.Minute, Field.AskHigh, "Ask.High."),
+            };
+
+            _optionIndicator = Identity(_option, Resolution.Minute, Field.Volume, "Volume.");
+        }
+
+        public override void OnData(Slice slice)
+        {
+            if (slice.QuoteBars.ContainsKey(_equity))
+            {
+                _quoteBarsFound = true;
+                var wrongEquityIndicators = _equityIndicators.Where(x =>
+                {
+                    var propertyName = x.Name.Split(".")[0]; // This could be Ask/Bid
+                    var secondPropertyName = x.Name.Split(".")[1]; // This could be Open/Close/High/Low
+                    var property = slice.QuoteBars[_equity].GetType().GetProperty(propertyName).GetValue(slice.QuoteBars[_equity], null);
+                    var value = (decimal)property.GetType().GetProperty(secondPropertyName).GetValue(property, null);
+                    return x.Current.Value != value;
+                });
+
+                if (wrongEquityIndicators.Any())
+                {
+                    throw new RegressionTestException();
+                }
+            }
+
+            if (slice.OptionChains.ContainsKey(_option.Canonical) && slice.OptionChains[_option.Canonical].TradeBars.ContainsKey(_option))
```

Comment:
Just a thought for good practices, perhaps doing `slice.OptionChains[_option.Canonical].TradeBars.TryGetValue(_option)` instead of `ContainsKey` + `.TradeBars[_option]`